pkg/metadata/system.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package metadata
+
+import (
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/prometheus/common/model"
+
+	"github.com/parca-dev/parca-agent/pkg/buildinfo"
+)
+
+var labels model.LabelSet
+
+func int8SliceToString(arr []int8) string {
+	var b strings.Builder
+	for _, v := range arr {
+		// NUL byte, as it's a C string.
+		if v == 0 {
+			break
+		}
+		b.WriteByte(byte(v))
+	}
+	return b.String()
+}
+
+func kernelRelease() (string, error) {
+	var uname syscall.Utsname
+	if err := syscall.Uname(&uname); err != nil {
+		return "", err
+	}
+
+	return int8SliceToString(uname.Release[:]), nil
+}
+
+// System provides metadata for the current system.
+func System() *Provider {
+	var once sync.Once
```

Comment:
I think this is unnecessary now that I have re-read the code. `System` only gets called once,j anyway.

Reply:
While this is true, and it will only be called once, it's because we only have one profiler right now. By memoizing the value here we ensure the value is never recomputed. I am thinking for example once we add multiple profilers or if the way we call the metadata providers and we start calling `System()` more than once

What do you think?

Reply:
If it's the intended design, then we need to move the sync.Once to package level.